code/game/turfs/unsimulated/beach.dm

```diff
@@ -34,8 +34,31 @@
 /turf/unsimulated/beach/water
 	name = "Shallow Water"
 	icon_state = "seashallow"
+	var/obj/machinery/poolcontroller/linkedcontroller = null
 	water_overlay_image = "water_shallow"
 
+/turf/unsimulated/beach/water/Entered(atom/movable/AM, atom/OldLoc)
+	. = ..()
+	if(!linkedcontroller)
+		return 
+	if(ishuman(AM))
+		var/mob/living/carbon/human/M = AM

```

Comment:
These are all pointless definitions. Typecasting in this way does literally nothing. BYOND tosses type data.